### PR TITLE
Limit core profiler extensions to 8

### LINKED ISSUE #53364: Add Printful to core profiler recommendations
## Issue Description

Add printful to the core profiler "Free business features" screen.

## Acceptance Criteria

* Use the following assets:
  * Icon: plugins/woocommerce/client/admin/client/task-lists/fills/products/icon/printful.png
  * Title: Print on-demand products
  * Description: Design and easily sell custom print products online with Printful.
* Display when "Clothing and accessories" industry selected on previous screen, possibly using an [options transformer](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Admin/RemoteSpecs/RuleProcessors/README.md#option-transformer).
* Display for the following countries: Australia, UK, Canada
* If not already in place, ensure this screen is limited to a maximum of 8 recommendations.
* Display in the last position (currently 8th for these countries), or not at all if the maximum recommendations is already reached.
* The Printful placement is experimental. Add it via  the WCCOM recommendations only so that to be easily changed if needed. Don't change the core fallback.

### PR DESCRIPTION
**Pull request by ilyasfoo**
### Changes proposed in this Pull Request:

Partially addresses #53364

- Adds 8 hard limit to number of core profiler extensions

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install and activate Code Snippets plugin
2. Activate the following code snippet: <br> 
```php
function filter_free_extensions_response($response) {
    $plugins = array();
    for($i = 1; $i <= 9; $i++) {
        $plugins[] = array(
            "name" => "WooPlugin " . $i,
            "description" => "Plugin " . $i . " description",
            "is_visible" => true,
            "is_built_by_wc" => true,
            "key" => "woo-plugin-" . strtolower($i),
            "label" => "Plugin " . $i . " Label",
            "image_url" => "https://woocommerce.com/wp-content/plugins/wccom-plugins/obw-free-extensions/images/core-profiler/logo-woo.svg",
            "learn_more_link" => "https://woocommerce.com",
            "install_priority" => $i,
            "is_installed" => false,
            "is_activated" => false
        );
    }

    return array(
        array(
            "key" => "obw/core-profiler",
            "title" => "Grow your store",
            "plugins" => $plugins
        )
    );
}

add_filter('rest_pre_echo_response', function($response, $object, $request) {
    if ($request->get_route() === '/wc-admin/onboarding/free-extensions') {
        return filter_free_extensions_response($response);
    }
    return $response;
}, 10, 3);
```
4. In JS console, run `localStorage.setItem( 'debug', 'wc-admin:*' );`
4. Go through core profiler until the extensions step
5. Observe it shows 8 extensions despite having 9 extensions
6. Click next to attempt installation
7. Observe the track `wcadmin_coreprofiler_store_extensions_continue` is fired with 1 truncated plugin

image

image

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/client/admin/client/core-profiler/pages/Plugins/components/plugin-error-banner/PluginErrorBanner.tsx [hunk 1]
```diff
@@ -25,7 +25,7 @@ export const PluginErrorBanner = ( {
 		case pluginsInstallationPermissionsFailure:
 		case pluginsInstallationErrors?.some(
 			// it really shouldn't get here since permissions are pre-checked. but we'll check for 403 just to be safe.
-			( e ) => e.errorDetails?.data.data.status === 403 // 403 is the code representing rest_authorization_required_code()
+			( e ) => e.errorDetails?.data?.data?.status === 403 // 403 is the code representing rest_authorization_required_code()
```
Comment (ilyasfoo) [inline plugins/woocommerce/client/admin/client/core-profiler/pages/Plugins/components/plugin-error-banner/PluginErrorBanner.tsx:28]: When it's not 403, such as cases when plugin slug is invalid, this fatals

File: plugins/woocommerce/client/admin/client/core-profiler/index.tsx [hunk 2]
```diff
@@ -626,7 +627,16 @@ const handlePlugins = assign( {
 	}: {
 		event: DoneActorEvent< Extension[] >;
 	} ) => {
-		return event.output;
+		return event.output.slice( 0, 8 );
```
Comment (rjchow) [inline plugins/woocommerce/client/admin/client/core-profiler/index.tsx]:
```suggestion
		return event.output.slice( 0, 8 ); // in lieu of a plugin display priority system, we're only showing the first 8 plugins in the recommendations list
```

File: plugins/woocommerce/client/admin/client/core-profiler/pages/Plugins/components/plugin-error-banner/PluginErrorBanner.tsx (hunk 1 above)
Comment (rjchow) [reply plugins/woocommerce/client/admin/client/core-profiler/pages/Plugins/components/plugin-error-banner/PluginErrorBanner.tsx:28]: oooooooh

**Review by rjchow — APPROVED** [review]
tested well, thanks!

**Pull request merged**